src/gt4sd/domains/materials/property_scorer.py

```diff
@@ -0,0 +1,178 @@
+        """
+
+        self.validate_input(sample=sample, property_name=self.name)
+        return self.get_distance(self.scoring_function(sample) - self.target)
+
+    def score_list(self, sample_list: List[str]) -> List[float]:
+        """Generates a list of scores for a given molecule or protein list.
+
+        Args:
+            samples_list: A List of molecules or proteins.
+
+        Returns:
+            A List of scores
+        """
+        scores = []
+        for sample in sample_list:
+            scores.append(self.score(sample))
+        return scores
+
+    def predictor(self, sample: str) -> Union[float, int]:
+        """Generates a prediction for a given molecule or protein.
+
+        Args:
+            sample: molecule or protein.
+
+        Returns:
+            A score for the given SMILES
+        """
+
+        self.validate_input(sample=sample, property_name=self.name)
+        return self.scoring_function(sample)
+
+    def validate_input(self, sample: str, property_name: str) -> None:
+        """Validates the sample in input.
+        If self.name is a property available for molecules, check that sample is a SMILES.
+        If self.name is a property available for proteins, check that sample is a protein.
+
+        Args:
+            sample: molecule or protein.
+
+        Raises:
+            ValueError: if the sample is not a valid SMILES or protein given a certain property.
+
+        Returns:
+            True if the input is valid.
+        """
+        # if selected property is available for molecules, check that sample is a SMILES
+        if self.name in MOLECULE_PROPERTY_PREDICTOR_FACTORY:
+            if Chem.MolFromSmiles(sample) is None:
+                raise ValueError(
+                    f"{property_name} is a property available for molecules and {sample} is not a valid SMILES. Please input a molecule."
+                )
+        else:
+            # if property is available for proteins, check that sample is a FASTA
+            if Chem.MolFromFASTA(sample) is None:
+                raise ValueError(
+                    f"{property_name} is a property available for proteins and {sample} is not a valid FASTA. Plese input a protein."
+                )
+
+
+class MoleculePropertyPredictorScorer(PropertyPredictorScorer):
+    """Property Predictor Scorer for molecules."""
+
+    def __init__(
+        self,
+        name: str,
+        target: Union[float, int],
+        parameters: Dict[str, Any] = {},
+    ) -> None:
+        """Scoring function that calculates a generic score for a property in molecules.
+
+        Args:
+            name: name of the property to score.
+            target: target score that will be used to get the distance to the score of a molecule or protein (not be confused with parameters["target"]).
+            parameters: parameters for the scoring function.
+        """
+        self.name = name
+        self.target = target
+        self.parameters = parameters
```

Comment:
I might be wrong but it looks like most of these attribute assignments are redundant